### PR TITLE
Update README for some M1 explanations

### PR DESCRIPTION
**Pull request by famleebob**
Add clarification, and some cover some issues uncovered doing my first build on an M1 Mac.  Expansions on the text are included as drop downs so to not complicate the general checklist.

Signed-off-by: Gerald (Bob) Lee <gerald@zededa.com>

### CODE REVIEW

**Comment by deitch** [comment]
Will review, but can you please squash the commits into a single one with a proper description? Thanks.

File: README.md [hunk 1]
```diff
@@ -84,6 +92,16 @@ This step is required on **Linux** and is required to create eve bootable images
 ```sh
 $ docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
 ```
+<details><summary>M1 Mac cross builds</summary>
+
+For docker to emulate a chip architecture different from the build host, perform this additional step:
```
Comment (deitch) [inline README.md:97]: Is this necessary? The previous step - `multiarch/qemu-user-static` - is meant to provide qemu user static. As for `binfmt`, docker desktop ships with binfmt already run.

**Comment by famleebob** [comment]
Quoting Avi Deitcher ***@***.***>:

> @deitch commented on this pull request.
>
>
>
>> @@ -84,6 +92,16 @@ This step is required on **Linux** and is  
>> required to create eve bootable images
>  ```sh
>  $ docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
>  ```
> +<details><summary>M1 Mac cross builds</summary>
> +
> +For docker to emulate a chip architecture different from the build  
> host, perform this additional step:
>
> Is this necessary? The previous step - `multiarch/qemu-user-static`  
> - is meant to provide qemu user static. As for `binfmt`, docker  
> desktop ships with binfmt already run.

I ran into a problem building on my M1 Mac.  Eric recalled needing
to do this (from the redit post), I applied the change, and it worked.

I believe it is only required in a cross build environment.  That is,
building for amd64 on an aarch64 machine.

> --
> Reply to this email directly or view it on GitHub:
> https://github.com/lf-edge/eve/pull/2817#pullrequestreview-1111911293
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>

-bob

**Comment by deitch** [comment]
I find it strangeI find it strange. I was able to run amd64 natively on docker without an issue.

Can you restart your docker engine (so you get a clean slate), and then see what happens when you just try to run something cross-arch, like:

```
docker run -it --rm --entrypoint=sh --platform linux/amd64 linuxkit/getty:v0.8-amd64
```

Then we can move on to the build. 
```

**Pull request closed without merging**